src/components/themed/SwapInput.tsx

```diff
@@ -1,190 +1,283 @@
+   * that we force the user to input a crypto amount, even if the caller wanted
+   * to initialize the focused flip input field with fiat.
+   */
+  const overrideForceField = useMemo(() => {
+    const cryptoCurrencyCode = getCurrencyCode(wallet, tokenId)
+    const fiatValue = convertCurrency('100', cryptoCurrencyCode, wallet.fiatCurrencyCode)
+    return fiatValue === '0' ? 'crypto' : forceField
+  }, [convertCurrency, forceField, tokenId, wallet])
 
-  if (props.isThinking) {
+  const renderHeader = () => {
     return (
-      <View style={[styles.container, styles.containerNoFee, styles.containerNoWalletSelected]}>
-        <View style={styles.topRow}>
-          <ActivityIndicator color={theme.iconTappable} />
-        </View>
-      </View>
+      <Header>
+        <CardHeading>{heading}</CardHeading>
+        <Space sideways>
+          <WalletPlaceHolder onPress={handleWalletPlaceholderPress}>
+            <WalletPlaceHolderText>{walletPlaceholderText}</WalletPlaceHolderText>
+          </WalletPlaceHolder>
+        </Space>
+      </Header>
     )
   }
 
-  if (wallet == null) {
-    return <MainButton label={props.buttonText} type="secondary" onPress={launchSelector} />
-  }
-
-  if (!props.isFocused) {
-    return (
-      <CardUi4>
-        <RowUi4 icon={<CryptoIconUi4 sizeRem={1.75} walletId={wallet.id} tokenId={tokenId} />} onPress={focusMe}>
-          <EdgeText style={styles.text}>{guiWalletName + ': ' + currencyCode}</EdgeText>
-        </RowUi4>
-      </CardUi4>
-    )
+  const renderIcon = () => {
+    return <CryptoIconUi4 marginRem={0} pluginId={wallet.currencyInfo.pluginId} sizeRem={1.75} tokenId={tokenId} />
   }
 
   return (
     <>
-      <EdgeText style={styles.errorText}>{errorMessage}</EdgeText>
-      {renderBalance()}
-      <CardUi4>
-        <ExchangedFlipInput2
-          onNext={onNext}
-          onFocus={props.onFocus}
-          onBlur={props.onBlur}
-          headerText={props.headerText}
-          headerCallback={launchSelector}
-          onAmountChanged={handleAmountsChanged}
-          startNativeAmount={overridePrimaryNativeAmount}
-          keyboardVisible={false}
-          forceField="fiat"
-          tokenId={tokenId}
-          wallet={wallet}
-        />
-        {children}
-      </CardUi4>
+      <FlipInput2
+        disabled={disabled}
+        onBlur={onBlur}
+        onFocus={onFocus}
+        onNext={onNext}
+        ref={flipInputRef}
+        convertValue={convertValue}
+        fieldInfos={fieldInfos}
+        renderHeader={renderHeader}
+        renderIcon={renderIcon}
+        returnKeyType={returnKeyType}
+        forceFieldNum={forceFieldMap[overrideForceField]}
+        inputAccessoryViewID={inputAccessoryViewID}
+        keyboardVisible={keyboardVisible}
+        startAmounts={[initialDisplayAmount, initialFiatAmount]}
+        placeholders={[lstrings.string_tap_to_edit, lstrings.string_tap_next_for_quote]}
```

Comment:
Since you sorted the props in `FlipInput2` itself, I was half-expecting them to be sorted here too.

Reply:
Ooh, good idea. I like sorting things 😉